draft-ietf-tls-esni.md

```diff
@@ -1829,6 +1829,35 @@ registration policy ({{!RFC8126, Section 4.6}}).
 The registration policy for for the "ECHConfig Extension Type" registry
 is Specification Required {{!RFC8126}}.
 
+This document defines several Reserved values for ECH configuration extensions.
+These can be used by clients and servers to "grease" the contents of the
+ECH configuration, as inspired by {{?RFC8701}}. Implementations SHOULD
+select reserved values at random when including them in ECH configurations.
+Note that the reserved values with the high-order bit set to 1 are mandatory,
+as defined in {{config-extensions}}.
```

Comment:
```suggestion
ECH configuration, as inspired by {{?RFC8701}}. This helps ensure clients
process ECH extensions correctly. When constructing ECH configurations,
servers SHOULD randomly select from reserved values with the high-order
bit clear. Correctly-implemented client will ignore those extensions.

The reserved values with the high-order bit set are mandatory, as defined
in {{config-extensions}}. Servers SHOULD randomly select from these
values and include them in extraneous ECH configurations. These
extraneous ECH configurations SHOULD have invalid keys, and public
names which the server does not respond to. Correctly-implemented
clients will ignore these configurations.
```

Reply:
I'd be fine with spelling stuff out for the ECHConfig generator. I dunno why "extraneous" is a SHOULD though, nor why invalid key values are better? I'd need to think whether or not inventing random public_name values is a good idea.

Reply:
Ah, I guess you want a random key so decryption definitely fails. That seems reasonable. Still not sure about a random public_name though. In any case, some text like this is needed and shows I guess that GREASEing that mandatory bit requires an additional ~70 octets.

Reply:
Junk public name is so we don't mask the decryption failure with the recovery flow. Don't think it needs to be random though. `ech.invalid` is probably good enough. :-)

No strong feelings about SHOULD or MAY or whatever. This is a bit heavier, so downgrading the shouty verb seems reasonable I suppose.